src/shared/syscheck_op.c

```diff
@@ -850,18 +850,24 @@ static int process_ace_info(void *ace, cJSON *acl_json) {
     }
 
     if (error = w_get_account_info(sid, &account_name, &domain_name), error) {
+        os_free(account_name);
+        os_free(domain_name);
         mdebug2("No information could be extracted from the account linked to the SID. Error: %d.", error);
+        return 1;
```

Comment:
In this case, we cannot return and free the variables since the event is generated even the account information is missing.